### PR TITLE
added logging of debug code execution with name and UID

### PR DESCRIPTION
**Pull request by HakonRydland**
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    Now logs debug console executions with the users in game name and UID, this includes logging the target: local, server, global, this allows server owners to track what code has been run on their servers to help track hackers, but also keeps a record of debug code run that may have caused a bug etc.

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Review by Lazejun — APPROVED** [review]
From a User side it seems to work.

Might come in handy when we have People in help channels saying Commands would not work when they used them wrong.

File: A3A/addons/JeroenArsenal/JND/fn_debugv2.sqf [hunk 1]
```diff
@@ -156,7 +157,15 @@ fnc_debugv2_overwrite = {
 					};
 
 					_input = (_strings joinString "") + _input;
-
+                    //logging of executed code for security
+                    call {
+                        if (A3A_logDebugConsole isEqualTo -1) exitWith {};
+                        if !(
```
Comment (jaj22) [inline A3A/addons/JeroenArsenal/JND/fn_debugv2.sqf:163]: This `!` shouldn't be here. It's currently causing it to log only devs for "All non-devs" and log no-one for "All".

File: A3A/addons/JeroenArsenal/JND/fn_debugv2.sqf [hunk 2]
```diff
@@ -277,8 +286,27 @@ fnc_debugv2_overwrite = {
 		((UiNameSpace getVariable "jn_debugConsole_buttons") # _index) ctrlSetText _name;
 	}];
 
-
-
+    //logging of executed code for security
+    if (A3A_logDebugConsole isEqualTo -1) exitWith {};
+    if !(
```
Comment (jaj22) [inline A3A/addons/JeroenArsenal/JND/fn_debugv2.sqf:291]: Also this `!`.

File: A3A/addons/JeroenArsenal/JND/fn_debugv2.sqf [hunk 3]
```diff
@@ -277,8 +286,27 @@ fnc_debugv2_overwrite = {
...
+            private _btnName = _ctrl getVariable ["JN_Debug_buttonName", "[ERROR NONAME]"];
+            private _code = ctrlText (findDisplay 49 displayCtrl 12284);
+            ServerInfo_4("Debug line executed as %1 by %2[%3]"+endl+"=============="+endl+"%4"+endl+"==============",_btnName,name player, getPlayerUID player, _code);
+        }];
+    } forEach [
+        [13285, "Global Exec"],
+        [13286, "Server Exec"],
+        [13484, "Local Exec"],
+        [13284, "Performance test"]
+    ];
```
Comment (jaj22) [inline A3A/addons/JeroenArsenal/JND/fn_debugv2.sqf:309]: This doesn't respond to changes in the A3A_logDebugConsole var after init. If this is intentional then it should also apply to the other section, which does respond to later changes. Make them consistent either way.

File: A3A/addons/maps/MissionDescription/params.hpp [hunk 4]
```diff
@@ -333,6 +333,13 @@ class Params
         texts[] = {"Load from save (Default: Debug)", "Error", "Info", "Debug"};
         default = 9999;
     };
+    class A3A_logDebugConsole
+    {
+        title = "Log debug console use";
+        values[] = {9999,-1,1,2};
+        texts[] = {"Load from save (Default: All none-dev)", "None", "All none-dev", "All"};
```
Comment (jaj22) [inline A3A/addons/maps/MissionDescription/params.hpp:340]: Should be "All non-dev" rather than "All none-dev".

**Review by jaj22 — CHANGES_REQUESTED** [review]

**Review by jaj22 — APPROVED** [review]

**Pull request merged**